src/main/java/org/threeten/extra/chrono/InternationalFixedChronology.java

```diff
@@ -0,0 +1,431 @@
+     * @throws ClassCastException if the {@code era} is not of the correct type for the chronology
+     */
+    @Override
+    public int prolepticYear (final Era era, final int yearOfEra) {
+        if (!(era instanceof InternationalFixedEra)) {
+            throw new ClassCastException ("Era must be InternationalFixedEra");
+        }
+
+        if (1 > yearOfEra) {
+            throw new DateTimeException ("Year of era MUST not be negative!");
+        }
+
+        return yearOfEra;
+    }
+
+    /**
+     * Gets the range of valid values for the specified field.
+     * <p>
+     * All fields can be expressed as a {@code long} integer.
+     * This method returns an object that describes the valid range for that value.
+     * <p>
+     * Note that the result only describes the minimum and maximum valid values
+     * and it is important not to read too much into them. For example, there
+     * could be values within the range that are invalid for the field.
+     * <p>
+     * This method will return a result whether or not the chronology supports the field.
+     *
+     * @param field the field to get the range for, not null
+     * @return the range of valid values for the field, not null
+     * @throws DateTimeException if the range for the field cannot be obtained
+     */
+    @Override
+    public ValueRange range (final ChronoField field) {
+        switch (field) {
+            case PROLEPTIC_MONTH:
+                return PROLEPTIC_MONTH_RANGE;
+            case YEAR_OF_ERA:
+                return YEAR_RANGE;
+            case YEAR:
+                return YEAR_RANGE;
+            case DAY_OF_MONTH:
+                return DAY_OF_MONTH_RANGE;
+            case DAY_OF_YEAR:
+                return DAY_OF_YEAR_RANGE;
+            case MONTH_OF_YEAR:
+                return MONTH_OF_YEAR_RANGE;
+            default:
+                return field.range ();
+        }
+    }
+
+    /**
+     * Resolve singleton.
+     *
+     * @return the singleton instance, not null
+     */
+    @SuppressWarnings ("static-method")
+    private Object readResolve () {
+        return INSTANCE;
+    }
+
+    //-----------------------------------------------------------------------
+
+    /**
+     * Obtains a local date in Julian calendar system from the
+     * era, year-of-era, month-of-year and day-of-month fields.
+     *
+     * @param era        the Julian era, not null
+     * @param yearOfEra  the year-of-era
+     * @param month      the month-of-year
+     * @param dayOfMonth the day-of-month
+     * @return the Julian local date, not null
+     * @throws DateTimeException if unable to create the date
+     * @throws ClassCastException if the {@code era} is not a {@code JulianEra}
+     */
+    @Override
+    public InternationalFixedDate date (final Era era, final int yearOfEra, final int month, final int dayOfMonth) {
+        return date (prolepticYear (era, yearOfEra), month, dayOfMonth);
```

Comment:
The current style/formatting doesn't put a space between the method name and the opening parenthesis.  Checkstyle should be catching this.